src/anm/events.js

```diff
@@ -164,28 +164,31 @@ MouseEventsSupport.prototype.dispatch = function(event) {
         return true;
     }
 
-    var dispatchedChild;  // found the matching child inside
-    if (owner.inside(localEvent)) {
-        owner.reverseEach(function(child) {
-            if (child.isActive()) {
-                dispatchedChild = child.dispatchMouseEvent(localEvent) ? child : null;
-                if (dispatchedChild) return false; // stop iteration of reverseEach
-            }
-        });
+    var dispatchedByOwner, // handled event myself
+        dispatchedByChild; // not handled myself, but found the matching child inside
 
-        if (!dispatchedChild) {
-            if (localEvent.type === 'mousemove') {
-                this.processMove(localEvent); // fire mouseenter/mouseexit if required
-            }
-            if ((localEvent.type === 'mouseclick') || (localEvent.type === 'mousedown')) {
-                state.pressedNode = this.owner;
-            }
-            this.owner.fire(localEvent.type, localEvent);
+    // here and below localEvent has properties `.x` and `.y`, so duck typing works
+    dispatchedByOwner = owner.inBounds && owner.inBounds(localEvent) && owner.inside && owner.inside(localEvent);
+    // scenes have no .inside or .inBounds methods
```

Comment:
of course scene is visual element. 
- it can paint itself
- it receives events
- it has bounds (0,0,width,height)
- it has children
- it has background
- we can apply transform to scene (camera)
- we can test events whether they are inside scene or not (yes, currently scene gets only events which canvas gets, i.e. all, but still)

in fact scene is almost the same as composite element (clip)